python_docs_theme/static/copybutton.js

```diff
@@ -1,64 +1,63 @@
-    var border_color = pre.css('border-top-color');
-    var button_styles = {
-        'cursor':'pointer', 'position': 'absolute', 'top': '0', 'right': '0',
-        'border-color': border_color, 'border-style': border_style,
-        'border-width': border_width, 'color': border_color, 'text-size': '75%',
-        'font-family': 'monospace', 'padding-left': '0.2em', 'padding-right': '0.2em',
-        'border-radius': '0 3px 0 0'
-    }
-
-    // create and add the button to all the code blocks that contain >>>
-    div.each(function(index) {
-        var jthis = $(this);
-        if (jthis.find('.gp').length > 0) {
-            var button = $('<span class="copybutton">&gt;&gt;&gt;</span>');
-            button.css(button_styles)
-            button.attr('title', hide_text);
-            button.data('hidden', 'false');
-            jthis.prepend(button);
-        }
-        // tracebacks (.gt) contain bare text elements that need to be
-        // wrapped in a span to work with .nextUntil() (see later)
-        jthis.find('pre:has(.gt)').contents().filter(function() {
-            return ((this.nodeType == 3) && (this.data.trim().length > 0));
-        }).wrap('<span>');
-    });
+    const hide_text = 'Hide the prompts and output';
+    const show_text = 'Show the prompts and output';
 
-    // define the behavior of the button when it's clicked
-    $('.copybutton').click(function(e){
-        e.preventDefault();
-        var button = $(this);
-        if (button.data('hidden') === 'false') {
+    const button = document.createElement("span")
+    button.classList.add("copybutton")
+    button.innerText = ">>>"
+    button.title = hide_text
+    button.dataset.hidden = false
+    const buttonClick = event => {
+        // define the behavior of the button when it's clicked
+        event.preventDefault();
+        const buttonEl = event.currentTarget
+        const codeEl = buttonEl.nextElementSibling
+        if (buttonEl.dataset.hidden === 'false') {
             // hide the code output
-            button.parent().find('.go, .gp, .gt').hide();
-            button.next('pre').find('.gt').nextUntil('.gp, .go').css('visibility', 'hidden');
-            button.css('text-decoration', 'line-through');
-            button.attr('title', show_text);
-            button.data('hidden', 'true');
+            codeEl.querySelectorAll('.go, .gp, .gt').forEach(el => el.hidden = true)
+            codeEl.querySelectorAll('.gt').forEach(el => {
+                while ((el = el.nextSibling) && el.nodeType !== Node.DOCUMENT_NODE) {
+                    if (el.nodeType === Node.ELEMENT_NODE && el.matches(".gp, .go")) break;
+                    if (el.nodeType === Node.TEXT_NODE && el.textContent.trim()) {
+                        const wrapper = document.createElement('span');
+                        el.after(wrapper);
+                        wrapper.appendChild(el);
+                        el = wrapper
+                    }
```

Comment:
Why is this necessary?

Reply:
I've added a note -- in efect tracebacks have bare text elements that we need to be able to hide, so we wrap in a span. This replicates the current behaviour.

A